src/Symfony/Component/Messenger/Transport/InMemory/InMemoryTransport.php

```diff
@@ -46,16 +48,25 @@ class InMemoryTransport implements TransportInterface, ResetInterface
     private array $queue = [];
 
     private int $nextId = 1;
-    private ?SerializerInterface $serializer;
+    private array $availableAt = [];
 
-    public function __construct(SerializerInterface $serializer = null)
-    {
-        $this->serializer = $serializer;
+    public function __construct(
+        private ?SerializerInterface $serializer = null,
+        private ?ClockInterface $clock = null,
+    ) {
     }
 
     public function get(): iterable
     {
-        return array_values($this->decode($this->queue));
+        $envelopes = [];
+        $now = $this->clock ? $this->clock->now() : new \DateTimeImmutable();
```

Comment:
`$now = $this->clock?->now() : new \DateTimeImmutable();`

Reply:
ok